docs/narratives/build_backends.md

```diff
@@ -0,0 +1,228 @@
+
+
+## Telling build backend to build a particular variant
+
+A wheel variant build is invoked through a build frontend (e.g., `pip`, `build`,
+`uv build`). There's multiple possible ways for the actual build (dependencies
+linked against, compiler flags, etc.) to be influenced:
+
+1. The plugin may set some standard environment variables (e.g., `CFLAGS`,
+   `CXXFLAGS`, `LDFLAGS`).
+2. The build backend may recognize a variant variable plus value and do
+   something specific to that (e.g., try detecting and linking a dependency).
+3. The user building the wheel may take actions to ensure the build does the
+   desired things to produce the wheel variant, e.g.:
+
+   - Provide the variant name and value
+   - Install required build-time dependencies before starting the build
+   - Apply patches to the package's source code before starting the build
+   - Modify environment variables
+   - Pass custom build flags to the backend via `--config-settings`
+   - Post-process the wheel, either with a tool like `auditwheel` or manually
+
+The simplest case is only (1). Invoking such a build for a package that respects
+`CFLAGS`/`LDFLAGS` and needs nothing else could look like:
+
+```bash
+# E.g., a default psabi-v3 usage with meson-python
+$ python -m build -Cvariant="psabi=x86-64-v3"
+```
+
+What actually happens under the hood for that build command? The build frontend,
+`build` in this case, only passes on the `-C` (shorthand for `--config-settings`)
+to the build backend. The build backend, `meson-python` in this case, may do the
+following:
+
+1. Query the plugins, find that `x86-plugin` provides the `psabi` variable, then
+   retrieve what that means in terms of `CFLAGS`/`CXXFLAGS`/`LDFLAGS` (that's
+   all one can probably support inside the plugin).
+2. Pass on those flags, either by amending those environment variables *or* by
+   translating them to CLI flags (e.g., C compiler flags can be passed like
+   `-Dc_args=xxx`, which is nicer for logging than amending environment
+   variables).
+3. Once the build is complete and the wheel is being assembled, modify the
+   wheel filename and modify the `METADATA` file to insert the variant content
+
+Now let's look at a more complicated example: building a NumPy wheel for
+`psabi=x86-64-v3`. The difference is that NumPy contains SIMD code and custom
+build flags to control whether and how that code gets included and used.
+The build backend cannot know this, hence the user invoking the build must
+instruct `meson-python` to only handle step (3) above, and not attempt to do
+(1) and (2). This is done by using the `-Cvariant-name` config setting, rather
+than `-Cvariant`:
+
+```bash
+# Build a NumPy psabi=x86-64-v3 wheel; this requires nonstandard flags (we want
+# to keep >v3 runtime-dispatched features)
+$ python -m build --wheel -Cvariant-name="psabi=x86-64-v3" -Csetup-args=-Dcpu-baseline=AVX2,FMA3 -Csetup-args=-Dcpu-dispatch=AVX512F,AVX512_SKX
```

Comment:
```suggestion
$ python -m build --wheel -Cvariant-name="x86_64::level::v3" -Csetup-args=-Dcpu-baseline=AVX2,FMA3 -Csetup-args=-Dcpu-dispatch=AVX512F,AVX512_SKX
```

Also, technically we've actually solved this via `-Cvariant`, with the variant being passed to setup args, so perhaps we ought to use a different example here.